front/src/components/map/Maplibre.vue

```diff
@@ -0,0 +1,171 @@
+                                    params: { query: config.query },
+                                })
+                            ).data;
+                            for (const feature of geojson.features) {
+                                const center = [
+                                    feature.bbox[0] +
+                                        (feature.bbox[2] - feature.bbox[0]) / 2,
+                                    feature.bbox[1] +
+                                        (feature.bbox[3] - feature.bbox[1]) / 2,
+                                ];
+                                const point = {
+                                    type: "Feature",
+                                    geometry: {
+                                        type: "Point",
+                                        coordinates: center,
+                                    },
+                                    place_name: feature.properties.display_name,
+                                    properties: feature.properties,
+                                    text: feature.properties.display_name,
+                                    place_type: ["place"],
+                                    center: center,
+                                };
+                                features.push(point);
+                            }
+                        } catch (e) {
+                            console.error(
+                                `Failed to forwardGeocode with error: ${e}`
+                            );
+                        }
+
+                        return {
+                            features: features,
+                        };
+                    },
+                };
+
+                const geocoder = new MaplibreGeocoder(geocoder_api, {
+                    maplibregl: maplibregl,
+                    flyTo: { duration: 0 },
+                    showResultsWhileTyping: false, // Autocomplete search not allowed by OpenStreetMap usage policy
+                });
+                this.map.addControl(geocoder);
+
+                geocoder.on("result", (e) => {
+                    const marker = new maplibregl.Marker({
+                        draggable: true,
+                        color: "red",
+                    })
+                        .setLngLat(e.result.center)
+                        .addTo(this.map);
+                    this.center = e.result.center;
+                    const markerEl = marker.getElement();
+                    markerEl.style.cursor = "pointer";
+                    marker.on("dragend", (e) => {
+                        this.center = Object.values(e.target.getLngLat());
+                    });
+                });
+            } catch (err) {
+                console.error("map error", err);
+            }
+        },
+    },
+};
+</script>
+
+<style scoped>
+.map-holder {
+    width: 100%;
+}
+#map {
+    height: 70vh;
+}
+</style>
```

Comment:
I'd make the map a little more responsive since most people use mobile phones nowadays.
This is how it's rendered on iPhone 6: 
<img width="364" alt="image" src="https://user-images.githubusercontent.com/31178790/156858812-ade8b51b-20a8-4ffb-996b-b6d896b161d6.png">

Reply:
Sorry for the lack of updates lately, the map was meant to be a popup on the form for users to select their address location. I see that the form has been removed. Is the geolocation API endpoint still needed now?